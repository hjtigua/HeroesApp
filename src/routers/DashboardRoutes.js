import React, { Fragment } from "react";
import { Redirect, Route, Switch } from "react-router";
import { DcScreen } from "../Components/dc/DcScreen";
import { HeroScreen } from "../Components/heroes/HeroScreen";
import { MarvelScreen } from "../Components/marvel/MarvelScreen";
import { SearchScreen } from "../Components/search/SearchScreen";
import { Navbar } from "../Components/ui/NavBar";

export const DashboardRoutes = () => {
  return (
    <Fragment>
      <Navbar />
      <div className="container mt-2">
        <Switch>
          <Route exact path="/marvel" component={MarvelScreen} />
          <Route exact path="/hero/:heroeId" component={HeroScreen} />
          <Route exact path="/dc" component={DcScreen} />
          <Route exact path="/search" component={SearchScreen} />
          <Redirect to="/marvel" />
        </Switch>
      </div>
    </Fragment>
  );
};
