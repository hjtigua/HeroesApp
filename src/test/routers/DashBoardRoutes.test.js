import { mount } from "enzyme";
import { MemoryRouter } from "react-router";
import { AuthContext } from "../../auth/AuthContext";
import { DashboardRoutes } from "../../routers/DashboardRoutes";
import React from "react";

describe("Pruebas en DashboardRoutes", () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: true,
      name: "Loko Carpio",
    },
  };

  test("debe mostrarse correctamente", () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".text-info").text().trim()).toBe("Loko Carpio");
  });
});
