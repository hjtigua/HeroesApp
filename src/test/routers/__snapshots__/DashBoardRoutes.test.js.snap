// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en DashboardRoutes debe mostrarse correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ms-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Loko Carpio
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-2"
  >
    <div
      className="row row-cols-1 row-cols-md-3 g-4 animate__animated animate__fadeIn"
    >
      <div
        className="col"
      >
        <div
          className="card"
        >
          <img
            alt="..."
            className="card-img-top"
            src="./assets/heroes/marvel-spider.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Spider Man
            </h5>
            <p
              className="card-text"
            >
              Peter Parker
               
              <a
                href="/hero/marvel-spider"
                onClick={[Function]}
              >
                Ver Más
              </a>
            </p>
          </div>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="card"
        >
          <img
            alt="..."
            className="card-img-top"
            src="./assets/heroes/marvel-captain.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Captain America
            </h5>
            <p
              className="card-text"
            >
              Steve Rogers
               
              <a
                href="/hero/marvel-captain"
                onClick={[Function]}
              >
                Ver Más
              </a>
            </p>
          </div>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="card"
        >
          <img
            alt="..."
            className="card-img-top"
            src="./assets/heroes/marvel-iron.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Iron Man
            </h5>
            <p
              className="card-text"
            >
              Tony Stark
               
              <a
                href="/hero/marvel-iron"
                onClick={[Function]}
              >
                Ver Más
              </a>
            </p>
          </div>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="card"
        >
          <img
            alt="..."
            className="card-img-top"
            src="./assets/heroes/marvel-thor.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Thor
            </h5>
            <p
              className="card-text"
            >
              Thor Odinson
               
              <a
                href="/hero/marvel-thor"
                onClick={[Function]}
              >
                Ver Más
              </a>
            </p>
          </div>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="card"
        >
          <img
            alt="..."
            className="card-img-top"
            src="./assets/heroes/marvel-hulk.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Hulk
            </h5>
            <p
              className="card-text"
            >
              Bruce Banner
               
              <a
                href="/hero/marvel-hulk"
                onClick={[Function]}
              >
                Ver Más
              </a>
            </p>
          </div>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="card"
        >
          <img
            alt="..."
            className="card-img-top"
            src="./assets/heroes/marvel-wolverine.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Wolverine
            </h5>
            <p
              className="card-text"
            >
              James Howlett
               
              <a
                href="/hero/marvel-wolverine"
                onClick={[Function]}
              >
                Ver Más
              </a>
            </p>
          </div>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="card"
        >
          <img
            alt="..."
            className="card-img-top"
            src="./assets/heroes/marvel-daredevil.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Daredevil
            </h5>
            <p
              className="card-text"
            >
              Matthew Michael Murdock
               
              <a
                href="/hero/marvel-daredevil"
                onClick={[Function]}
              >
                Ver Más
              </a>
            </p>
          </div>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="card"
        >
          <img
            alt="..."
            className="card-img-top"
            src="./assets/heroes/marvel-hawkeye.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Hawkeye
            </h5>
            <p
              className="card-text"
            >
              Clinton Francis Barton
               
              <a
                href="/hero/marvel-hawkeye"
                onClick={[Function]}
              >
                Ver Más
              </a>
            </p>
          </div>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="card"
        >
          <img
            alt="..."
            className="card-img-top"
            src="./assets/heroes/marvel-cyclops.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Cyclops
            </h5>
            <p
              className="card-text"
            >
              Scott Summers
               
              <a
                href="/hero/marvel-cyclops"
                onClick={[Function]}
              >
                Ver Más
              </a>
            </p>
          </div>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="card"
        >
          <img
            alt="..."
            className="card-img-top"
            src="./assets/heroes/marvel-silver.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Silver Surfer
            </h5>
            <p
              className="card-text"
            >
              Norrin Radd
               
              <a
                href="/hero/marvel-silver"
                onClick={[Function]}
              >
                Ver Más
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
